blogs/posts/2024-01-16_nearest_neighbour.qmd

```diff
@@ -0,0 +1,181 @@
+
+```{r}
+
+no_need <- practices |>
+  filter(has_orig_need == 0)
+
+with_need <- practices |>
+  filter(has_orig_need == 1)
+
+
+neighbours <- no_need |>
+  select(no_need_postcode = postcode,no_need_prac_code=practice_code) |>
+  st_join(with_need, st_is_within_distance, 1500) |>
+  st_drop_geometry() |>
+  select(id, no_need_postcode,no_need_prac_code) |>
+  inner_join(x = with_need, by = join_by("id")) 
+
+
+```
+
+```{r}
+#| echo: false
+#| warning: false
+#| message: false
+
+
+leaflet(neighbours) |> 
+  addTiles() |>
+  addCircleMarkers(color = "purple") |>
+  addMarkers(  -1.50686326666667, 52.4141089666667, popup = "Practice with no data"
+) |>
+  addCircles(-1.50686326666667, 52.4141089666667,radius=1500) |>
+  addMarkers(-1.46927, 52.51899, popup = "Practice with no data"
+) |>
+addCircles(-1.46927, 52.51899,radius=1500)
+```
+
+The data for the "neighbours" was grouped by the practice code of those without need data and a mean value calculated for each practice to generate an estimated value.
+
+```{r}
+
+no_need_neighbours_estimate <- neighbours |>
+  group_by(no_need_prac_code) |>
+    summarise(need_est=mean(value)) |>
+    st_drop_geometry(select(no_need_prac_code,need_est)) 
+
+
+
+```
+
+The original data was joined back to the "neighbours".
+
+```{r}
+
+  no_need_with_neighbours_estimate <- practices |>
+    left_join(no_need_neighbours_estimate, join_by(practice_code==no_need_prac_code))  |>
+    st_drop_geometry(select(practice_code,need_est))
+```
+
+```{r}
+#| echo: false
+#| warning: false
+#| message: false
+  no_need_with_neighbours_estimate |>
+    select(-has_orig_need,-id) |>
+    gt()
+
+```
+
+An updated dataframe was created of the need data using the actual need for the practice where available, otherwise using estimated need.
+
+```{r}
+#| include: false
+no_need_with_neighbours_estimate |>
+    mutate(need_to_use = case_when(value>=0 ~ value,
+                                       .default = need_est)) |>
+    select(practice_code,need_to_use) 
+  
+```
+
+```{r}
+#| echo: false
+#| warning: false
+#| message: false
+no_need_with_neighbours_estimate |>
+    mutate(need_to_use = case_when(value>=0 ~ value,
+                                       .default = need_est)
+    ) |>
+    select(practice_code,need_to_use) |>
+    gt()
+```
+
+For my project, this method has successfully generated a prevalence for 125 of the 151 practices without a need value, leaving just 26 practices without a need. This is using a 1.5 km radius. In each use case there will be a decision to make regarding a more accurate estimate (smaller radius) and therefore fewer matches versus a less accurate estimate (using a larger radius) and therefore more matches.
+
+This approach could be replicated for other similar uses/purposes.
```

Comment:
Can you give an example (health-related or otherwise)?

Reply:
Are you rferring to my final statement when you say can I give an example? Are you suggesting I give anonther example for its re-use besides the one I have already talked about?

Reply:
Yeah, thought it might be worth mentioning some other potential implementations if you had thought of any, but you can also leave that to the reader's imagination.

Reply:
Done